Update Lighthouse/Shipyard to devel

dfarrell07 opened the pull request:
See commit messages for details.

This update should fix Lighthouse's CI (nginx image breakage).

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

tpantelis commented:
I was trying this locally a few days ago. Because of the `logr` version bump, we need to bump `sigs.k8s.io/controller-runtime` to _v0.11.2_. However this leads to other build errors that should be fixed by https://github.com/submariner-io/admiral/pull/342. Hopefully that will be it.

mkolesnik commented on go.mod:
```diff
@@ -10,14 +10,13 @@ require (
 	github.com/fsnotify/fsnotify v1.5.1 // indirect
 	github.com/go-errors/errors v1.2.0 // indirect
-	github.com/go-logr/logr v0.4.0
+	github.com/go-logr/logr v1.2.0
```
This gets bumped which I guess isn't backward compatible :/

dfarrell07 committed:
Update go.mod to get latest Shipyard

Automated result of

go get github.com/submariner-io/shipyard@devel
go mod tidy

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 committed:
Update go.mod to get latest Ligthouse

Automated result of:

go get github.com/submariner-io/lighthouse
go mod tidy

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

dfarrell07 commented:
```
[12:52:59.764] [submariner-operator]$ CGO_ENABLED=0 go build -trimpath -ldflags -s -w -X github.com/submariner-io/submariner-operator/pkg/version.Version=devel-838eff9 		   -X github.com/submariner-io/submariner-operator/api/submariner/v1alpha1.DefaultSubmarinerOperatorVersion=devel -o bin/subctl-devel-838eff9-linux-amd64 ./pkg/subctl/main.go
# sigs.k8s.io/controller-runtime/pkg/log
vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:163:2: cannot use res (type *DelegatingLogger) as type logr.Logger in return argument
vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:179:2: cannot use res (type *DelegatingLogger) as type logr.Logger in return argument
vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:195:2: cannot use res (type *DelegatingLogger) as type logr.Logger in return argument
vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:67:26: cannot use NullLogger{} (type NullLogger) as type logr.Logger in argument to Log.Fulfill
vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:82:41: cannot use NullLogger{} (type NullLogger) as type logr.Logger in argument to NewDelegatingLogger
vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:86:6: cannot use Log (type *DelegatingLogger) as type logr.Logger in assignment
vendor/sigs.k8s.io/controller-runtime/pkg/log/log.go:88:13: assignment mismatch: 1 variable but logr.FromContext returns 2 values
vendor/sigs.k8s.io/controller-runtime/pkg/log/null.go:30:5: cannot use NullLogger{} (type NullLogger) as type logr.Logger in assignment
vendor/sigs.k8s.io/controller-runtime/pkg/log/null.go:48:32: undefined: logr.InfoLogger
vendor/sigs.k8s.io/controller-runtime/pkg/log/null.go:54:2: cannot use log (type NullLogger) as type logr.Logger in return argument
vendor/sigs.k8s.io/controller-runtime/pkg/log/null.go:54:2: too many errors
make: *** [Makefile:192: bin/subctl-devel-838eff9-linux-amd64] Error 2
```

dfarrell07 commented:
```
controllers/test/driver.go:56:17: invalid operation: cannot compare d.Client == nil (operator == not defined on untyped nil) (typecheck)
	if d.Client == nil {
	               ^
controllers/servicediscovery/servicediscovery_controller_test.go:46:12: t.DoReconcile undefined (type *testDriver has no field or method DoReconcile) (typecheck)
		_, _ = t.DoReconcile()
		         ^
controllers/servicediscovery/servicediscovery_controller_test.go:53:33: t.InitClientObjs undefined (type *testDriver has no field or method InitClientObjs) (typecheck)
				t.InitClientObjs = append(t.InitClientObjs, newDNSConfig(""), newDNSService(clusterIP))
				                            ^
<many more>
```

mkolesnik commented:
As you can see `controller-runtime` is pinned to an older version, it probably needs to be bumped as well

```
sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.9.0
```

dfarrell07 commented:
This is being done in https://github.com/submariner-io/submariner-operator/pull/2037
